apps/server/src/modules/school/uc/school.uc.ts

```diff
@@ -27,6 +27,18 @@ export class SchoolUc {
 			action: Actions.read,
 			requiredPermissions: [Permission.SCHOOL_EDIT],
 		});
+		const school: SchoolDO = await this.schoolService.getSchoolById(schoolId);
+
+		const shouldRestartMigration =
+			!!school.oauthMigrationFinished &&
+			oauthMigrationPossible &&
+			oauthMigrationMandatory === !!school.oauthMigrationMandatory &&
+			!oauthMigrationFinished;
+
+		if (shouldRestartMigration) {
+			await this.schoolMigrationService.restartMigration(schoolId);
+		}
+
 		const migrationDto: OauthMigrationDto = await this.schoolService.setMigration(
```

Comment:
The double type definition is not needed. OauthMigrationDto is already set as return type.